astropy/wcs/wcs.py

```diff
@@ -714,12 +716,28 @@ def _fix_scamp(self):
         SIP distortion parameters.
 
         See https://github.com/astropy/astropy/issues/299.
+
+        SCAMP uses TAN projection exclusively. The case of CTYPE ending
+        in -TAN should have been handled by ``_fix_pre2012_scamp_tpv()`` before
+        calling this function.
         """
-        # Nothing to be done if no WCS attached
         if self.wcs is None:
             return
 
-        # Nothing to be done if no PV parameters attached
+        # Delete SIP if CTYPE explicitly has '-TPV' code:
+        ctype = [ct.strip().upper() for ct in self.wcs.ctype]
+        if sum(ct.endswith("-TPV") for ct in ctype) == 2:
+            if self.sip is not None:
```

Comment:
Not a big deal, but for speed it would make more sense to start with this check if `sip` is present at all before looking for `-TPV`.

Reply:
This is a good observation. Will do soon.

Reply:
Upon reviewing the code, I think the order is important and so this suggestion will not work.

Reply:
I don't understand the wcs code well enough to know indeed.  Can the presence of `.sip` depend on accessing `self.wcs.ctype`?

Anyway, it was similar and am completely happy to just get it in!